config/sources/families/meson-s4t7.conf

```diff
@@ -126,6 +127,13 @@ function armbian_kernel_config__add_khadas_common_drivers() {
 
 		# Hack to ensure that we will recreate kernel if common_drivers source changes
 		kernel_config_modifying_hashes+=( "khadas_common_drivers=${GIT_INFO_COMMON_DRIVERS[SHA1]}" )
+	else
+		# Unfortunately, debug kernel has to be enabled to make khadas_common_drivers compile
+		display_alert "Enabling kernel config" "DEBUG_KERNEL=y, EXPERT=y, EMBEDDED=y to fix khadas_common_drivers compilation" "info"
+		kernel_config_modifying_hashes+=("CONFIG_DEBUG_KERNEL=y" "CONFIG_EXPERT=y" "CONFIG_EMBEDDED=y")
```

Comment:
This line needs to be in the if block. The modification of kernel_config_modifying_hashes is only needed during first call to calculating kernel artifact version string. As I understand, It serves no purpose whatsoever during second call.

Still as these configs are already set, feel free to ignore this comment. But keep it in mind for the future.